member.php

```diff
@@ -1599,8 +1600,29 @@
 			$password_length = 8;
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('update');
+
+		$userhandler->set_data(array(
+			'uid'		=> $user['uid'],
+			'username'	=> $user['username'],
+			'email'		=> $user['email'],
+			'password'	=> random_str($password_length)
+		));
+
+		if(!$userhandler->validate_user())
+		{
+			error($lang->error_unknownerror);

```

Comment:
With the previous version of code they were likely (because username and emails were verified too). Now they're very unlikely because only this can happen after the complex password issue will be fixed: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/user.php#L182

Anyways, now that you display the error `error($lang->error_unknownerror);` and the lang string can be removed because it will never happen (false is only returned with an error).

And the indendation is icorrent (spaces instead of tabs).